modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -156,6 +157,16 @@ export async function loadData(
 		);
 	}
 
+	if (Liferay.FeatureFlags['LPD-25230'] && parameters) {
+		const parametersArray = parameters.split('&');
+
+		parametersArray.forEach((parameter) => {
+			const [key, value] = parameter.split('=');
+
+			url.searchParams.append(key, value);
```

Comment:
A small nit: we need to ensure that `append()` is enough for our purposes. In other words, check there's no need to "merge" values of a parameter both managed by the FDS/DSM (like `nestedFields`, see [here](https://github.com/liferay/liferay-portal/blob/3954fa615adb3574dfa76dec7bc962da5d2d36c1/modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/fragment/renderer/FDSAdminFragmentRenderer.java#L861-L869)) and also included in the list of parameters manually added.

The use of `parameters` is likely to produce these "clashing" situation where FDS wants to set a filter but also parameter contain a `filter` already. 

By using `append()` we would include 2 separate `filter` query parameters, I'm not sure how endpoints will behave.

It's clear that this is an advanced setting, and thus we assume user "knows what (s)he is doing", however, if adding an initial filter to the fetched data prevents data set to include any other filter, the feature would look too limiting imho

@herrardo I'd say we could at least analyze current behavior for `filter` and `nestedFields`. Makes sense?

Reply:
I pushed up additional changes to merge the `sort`, `filter`, and `nestedFields` parameters.

For the filters I was trying to ignore the existing fields used, but it was very tricky because of the different ways a filter could be constructed. Plus I think it might cause very unexpected things by modifying the filter being attempted to pass in. I currently have the filter added on with an `and`. Do you see any issues with this approach?

https://github.com/liferay-frontend/liferay-portal/blob/093c11eff4249836bbe07e57a08761e3b2d6320d/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js#L174-L177

Reply:
I am requesting using different filter combinations such as:

`(alternateName eq 'thomas') and (alternateName eq 'test')`

`(alternateName eq 'thomas') and (givenName eq 'test')`

`(alternateName eq 'thomas') and (givenName eq+'Thomas')`

to understand is applies correctly the inclusion/exclusion using the same field or different and seems to work just fine.

Reply:
I'm assuming there is no possibility to pass 2 different `filter` parameters in the URL and magically expect the endpoint to obey both of them as the filter expressions need to be connected with a logical operator which is missing.

In that line of thought, I guess we have no choice but attempt some sort of "logical merging" in the filter due to the requirements we have to solve. This would entail:

- Having one single filter expression all the time
- Connect the expression provided in the parameters to the filter expression set by the filters

Said that, the implementation details may vary. Please note we're already building an odata expression out from a  combined filter expression [including possible previous values for the filter parameter](https://github.com/liferay/liferay-portal/blob/56a7c977f0e1d64b668de3cb8e1e631aac1e975f/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js#L132-L137). So we might be reinventing the wheel here.

Anyways, this way of connecting is something we're already doing when more than 1 filter is applied at the same time so this looks like the way to go in this case.